OptimizelySDK/Notifications/NotificationCenter.cs

```diff
@@ -0,0 +1,197 @@
+﻿/* 
+ * Copyright 2017, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using OptimizelySDK.Entity;
+using OptimizelySDK.Event;
+using OptimizelySDK.Logger;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OptimizelySDK.Notifications
+{
+    public class NotificationCenter
+    {
+        /// <summary>
+        /// Enum representing notification types.
+        /// </summary>
+        public enum NotificationType
+        {
+            Decision, // Activate called.
+            Track, // Event tracked.
+            FeatureAccess // Feature accessed
+        };
+
+        public delegate void DecisionCallback(Experiment experiment, string userId, UserAttributes userAttributes, Variation variation, LogEvent logEvent);
+
+        public delegate void TrackCallback(string eventKey, string userId, UserAttributes userAttributes, EventTags eventTags, LogEvent logEvent);
+
+        public delegate void FeatureAccessCallback(string featureKey, string userId, UserAttributes userAttributes, Variation variation);
+        
+        private ILogger Logger;
+
+        // Notification Id represeting number of notifications.
+        public int NotificationId { get; private set; } = 1;
+
+        // Associative array of notification type to notification id and notification pair
+        private Dictionary<NotificationType, Dictionary<int, object>> Notifications = 
+            new Dictionary<NotificationType, Dictionary<int, object>>();
+
+        public int NotificationsCount
+        {
+            get
+            {
+                int notificationsCount = 0;
+                foreach (var notificationsMap in Notifications.Values)
```

Comment:
I am not sure but can't we do using linq in a single statement. 
notificationMap.Sum(o => o.values.count) something like this.

Reply:
Ok, I will do that ... after learning LINQ :P

Reply:
ok then learn LINQ, Lambda, Action, Func, Predicates, etc...